### DISCUSSION TITLE
Bump in the wire firewall

### DISCUSSION THREAD

**Discussion by fkautz** [lfn-cnti/bestpractices#116] [General]
Use case: A "hello world" bump in the wire firewall CNF.

ref: https://docs.google.com/document/d/1gf9liRVDPXTN7SFfAnI9nfyi4kOjkTeF7Gik_XVh63g/edit

The bump in the wire firewall represents a “hello world” CNF which represents the “Bump in the Wire” pattern. For this example, we call out five variations:
- All Local
  - Local K8s cluster: all
  - External: none
- External Network Service
  - Local K8s cluster: client workload, firewall
  - External: network service
- Internal Network Service
  - Local K8s cluster: firewall, network service
  - External: client workload
- Infrastructure
  - Local K8s cluster: firewall
  - External: client workload, network service
- Consumer
  - Local K8s cluster: client workload, network service
  - External: firewall

All five follow the same pattern:
![diagram showing client workload, transparent firewall cnf, and network service in a line](https://user-images.githubusercontent.com/26697/112869767-202f0500-9083-11eb-847c-242e4943fdbb.png)

Important features about this use case:
- In the dataplane, the firewall is fully transparent
- The Firewall may have an IP for control. For purposes of this use case, it is not addressable through the data plane.

![diagram showing logical and detailed view of firewall. logical view has client workload with ip 192.0.2.2 connected with a horizontal line to network service with ip 192.0.2.1. the firewall is in grey above the line with a dotted line around the box.  In the detailed view the client workload is connected to a box with the text transparent firewall. The transparent firewall has a connection going up, with IP 192.51.100.2 to the right, going to a box with the text Firewall controller with ip 192.51.100.1](https://user-images.githubusercontent.com/26697/112869858-3a68e300-9083-11eb-9f1c-b042ad72305e.png)

Related use cases:
- There are several variations of bump in the wire such as IDS, DLP, network monitoring, billing/charging enforcement.
- Likewise, there are L2 variations which also exist. Most notable are ARP responders.

**Comment by iawells** [comment]
OK, so the 'use cases' appear to be design patterns to solve a requirement rather than requirements, at first blush.  That said, if you can make an argument that an end user is being pushed towards one of them for any specific reason (relating to humans or external factors, not 'this code is pretty') I'd let you off.

**Reply by taylor** [reply]
@iawells What are you referring to with "pushed towards one of them".  Are you saying to pick a specific use case and provide end user information?  I'm not clear what you are asking.
